indico/modules/events/contributions/controllers.py

```diff
@@ -34,4 +56,55 @@ class RHManageContributions(RHManageContributionsBase):
     """Display contributions management page"""
 
     def _process(self):
-        return WPManageContributions.render_template('management/contributions.html', self.event)
+        contribs = Contribution.find_all(event_id=self.event.id, is_deleted=False)
+        sessions = Session.find_all(event_id=self.event.id, is_deleted=False)
+        tracks = self.event.getTrackList()
+        return WPManageContributions.render_template('management/contributions.html', self.event, event=self.event,
+                                                     contribs=contribs, sessions=sessions, tracks=tracks)
+
+
+class RHManageContributionBase(RHManageContributionsBase):
+    """Base class for a specific contribution"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.contrib
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageContributionsBase._checkParams(self, params)
+        self.contrib = Contribution.get(request.view_args['contrib_id'])
+
+
+class RHManageContributionCreate(RHManageContributionsBase):
+    def _process_GET(self):
+        form = ContributionForm()
+        return jsonify_template('events/contributions/management/contrib_dialog.html', form=form)
+
+    def _process_POST(self):
+        form = ContributionForm()
+        if form.validate_on_submit():
+            contrib = create_contrib(self.event, form.data)
+            flash(_("Contribution '{}' created successfully").format(contrib.title), 'success')
+            return jsonify_data(html=_render_contribution_list(self.event))
+
+
+class RHManageContributionUpdate(RHManageContributionBase):
+    def _process_GET(self):
+        form = ContributionForm(obj=FormDefaults(self.contrib))
+        return jsonify_template('events/contributions/management/contrib_dialog.html', form=form)
+
+    def _process_POST(self):
+        form = ContributionForm()
+        if form.validate_on_submit():
+            update_contrib(self.contrib, form.data)
```

Comment:
i'd go for `update_contribution` and `self.contribution` - I'd only use the short form in places where saving the space is actually useful